### ISSUE TITLE
Improve the performance of the REST-API

### ISSUE THREAD

**Issue by Marko-ilmkreis** [hzi-braunschweig/SORMAS-Project#7905]
### Feature Description

The Rest API was designed to access data from the App on mobile phones/tablets.

Currently, the REST API is used in Germany to upload a large amount of cases/contacts to Sormas, as otherwise case processing is not possible. External software extensions to Sormas (Climedo, CISS) are used for this purpose. 

Unfortunately, we have noticed that the performance of the REST API is not designed for this use. Sometimes the applications do not receive a response and therefore load the cases/contacts multiple times into Sormas. This is due to the privacy-friendly design of the applications themselves not storing any data.

According to Netzlink, this is something that needs to be given priority and adjusted in the development of Sormas, as the errors occur even on servers that have already been allocated more resources.

**Comment by JonasCir** [comment]
@Marko-ilmkreis I think the rest of your issue is missing? :)

**Comment by JonasCir** [comment]
Agreed, I witnessed slow performance of the API myself a couple of times while running tasks locally.

I will not make a promise that I cannot keep, nevertheless #7813 might be a candidate which could lead to a better overall application performance. It is not a silver bullet, but it seems like a good start :)

While working on it, however, I noticed a couple of issues:
1. Performance testing is hard
2. We need to improve our developer tooling around this topic (minimum #7835) such that we have direct feedback for single changes
3. In general, we need to base our development here on real measuring and data:
   1. We need more _performance_ metrics/insights from real deployments. Recently, Netzlink included [Glowroot](https://glowroot.org/), but we need to find a way to make the insights available to the devs (@fhauptmann)
   1. I'm currently lacking insights into questions like: What endpoints are used frequently? Which of them are especially slow? What is the workload use there? etc. Only with this questions being answered, we will be able to work systematic on this topic. Something like [Grafana Tempo](https://grafana.com/docs/tempo/latest/) or similar would be awesome, but I guess it would require some support in SORMAS. Again @fhauptmann likely can provide some insights here.

**Comment by JonasCir** [comment]
@syntakker  could you follow up on this?
